site/en/docs/web-platform/bfcache-notrestoredreasons/index.md

```diff
@@ -0,0 +1,220 @@
+  ],
+  id: "",
+  name: "",
+  reasons: [],
+  src: "",
+  url:"a.com"
+}
+```
+
+If multiple cross-origin frames have blocking reasons, we randomly select one cross-origin iframe and report whether it blocked bfcache or not. For the rest of the frames, we report `null` for the `blocked` value. This is to stop bad actors from inferring information about user state on sites they don't control by embedding multiple third-party frames into a page and then comparing the blocking information from each.
+
+```js
+{
+  blocked: false,
+  children: [
+    /* cross-origin frames */
+    {url: "", src: "b.com", id: "b", name: "b", blocked: null, reasons: [], children: []},
+    {url: "", src: "c.com", id: "c", name: "c", blocked: true, reasons: [], children: []},
+    {url: "", src: "d.com", id: "d", name: "d", blocked: null, reasons: [], children: []}
+  ]
+  id: "",
+  name: "",
+  reasons: [],
+  src: "",
+  url:"a.com"
+}
+```
+
+See the [Security and privacy](https://github.com/rubberyuzu/bfcache-not-retored-reason/blob/main/NotRestoredReason.md#security-and-privacy) section in the explainer for more details about security and privacy considerations.
+
+## Blocking reasons
+
+As we said earlier, there are many different reasons why blocking could occur. We have compiled a [handy spreadsheet](https://docs.google.com/spreadsheets/d/1li0po_ETJAIybpaSX5rW_lUN62upQhY0tH4pR5UPt60/edit#gid=0) showing all the reason strings and explaining what they mean.
+
+There are a few major categories of reason that are worth calling out:
+
+* `Circumstantial`: This refers to blocking reasons not directly related to the developer's page code. For example a related component crashed, something went wrong with the loading process, the page is in a temporary state that can't be cached, bfcache disabled due to insufficient memory, or a service worker did something to the page that disqualifies it from being cached. 
+* `Extensions`: There are a few different reason messages related to extensions. Generally we combine quite a few different reasons into the "Extensions" reason. We are intentionally vague about extension-related blocking reasons because we don't want to give away too much information about what extensions the user have installed, which ones are active on the page, what they are doing, etc.
+* `PageSupportNeeded`: The developer's code is using a web platform feature that is otherwise not bfcache blocking, but it is currently in a state that is bfcache blocking. For example, the page currently has a [BroadcastChannel](https://developer.mozilla.org/docs/Web/API/BroadcastChannel) with registered listeners, or an open [IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API) connection.
+* `Unload Handler`: The page has registered an [`unload` handler](https://developer.mozilla.org/docs/Web/API/Window/unload_event), which currently [prevents the bfcache being used in some browsers](https://web.dev/bfcache/#never-use-the-unload-event).
```

Comment:
Just realised this is an instance of the `PageSupportNeeded` category

```suggestion
* `PageSupportNeeded`: The developer's code is using a web platform feature that is otherwise not bfcache blocking, but it is currently in a state that is bfcache blocking. For example, the page currently has a [BroadcastChannel](https://developer.mozilla.org/docs/Web/API/BroadcastChannel) with registered listeners, or an open [IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API) connection. Or the page has registered an [`unload` handler](https://developer.mozilla.org/docs/Web/API/Window/unload_event), which currently [prevents the bfcache being used in some browsers](https://web.dev/bfcache/#never-use-the-unload-event).
```